clippy_lints/src/redundant_closure_call.rs

```diff
@@ -62,9 +62,9 @@ impl<'tcx> Visitor<'tcx> for ReturnVisitor {
 
 /// Checks if the body is owned by an async closure
 fn is_async_closure(body: &hir::Body<'_>) -> bool {
-    if let hir::ExprKind::Closure(closure) = body.value.kind
-        && let [resume_ty] = closure.fn_decl.inputs
-        && let hir::TyKind::Path(hir::QPath::LangItem(hir::LangItem::ResumeTy, ..)) = resume_ty.kind
+    if let hir::ExprKind::Closure(_) = body.value.kind
+        && let Some(kind) = body.generator_kind
+        && kind == GeneratorKind::Async(AsyncGeneratorKind::Closure)
```

Comment:
Could we pull those statements together? As in

```suggestion
        && let Some(GeneratorKind::Async(AsyncGeneratorKind::Closure)) = body.generator_kind
```